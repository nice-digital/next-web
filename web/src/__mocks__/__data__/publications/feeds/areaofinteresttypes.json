{"_links":{"self":[{"href":"/feeds/areaofinteresttypes"}]},"_embedded":{"nice.publications:area-of-interest-type-list":{"_links":{"self":[{}]},"_embedded":{"nice.publications:area-of-interest-type":[{"_links":{"self":[{}]},"ETag":null,"LastModified":"2021-02-23T00:00:00","Enabled":true,"Name":"Antimicrobial prescribing","PluralName":"Antimicrobial prescribing","IdentifierPrefix":"AMP"},{"_links":{"self":[{}]},"ETag":null,"LastModified":"2021-02-23T00:00:00","Enabled":true,"Name":"Covid-19","PluralName":"Covid-19","IdentifierPrefix":"COV"}]},"ETag":null}},"ETag":null,"LastModified":"2021-02-23T00:00:00"}