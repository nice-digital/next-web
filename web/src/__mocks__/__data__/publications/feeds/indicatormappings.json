{
  "LastModified": "2022-12-01T11:15:30.9024748",
  "Mappings": [
      {
          "Url": "https://www.nice.org.uk/standards-and-indicators/qofindicators/product-not-found",
          "Id": "ABC123"
      },
      {
          "Url": "https://www.nice.org.uk/standards-and-indicators/qofindicators/the-percentage-of-patients-with-diabetes-with-a-record-of-testing-of-foot-sensation-using-a-10g-monofilament-within-the-preceding-12-months",
          "Id": "IND102"
      },
      {
          "Url": "https://www.nice.org.uk/standards-and-indicators/ccgoisindicators/coronary-heart-disease-ccg10",
          "Id": "IND7"
      },
      {
          "Url": "https://www.nice.org.uk/standards-and-indicators/qofindicators/the-percentage-of-patients-with-one-or-more-of-the-following-conditions-chd-atrial-fibrillation-chronic-heart-failure-stroke-or-tia-diabetes-or-dementia-with-a-fast-score-of-3-or-audit-c-score-of-5-in-the-preceding-2-years-who-have-received-brief-interven",
          "Id": "IND1001"
      },
      {
          "Url": "https://www.nice.org.uk/standards-and-indicators/ccgoisindicators/proportion-of-patients-with-atrial-fibrillation-not-on-anticoagulation-admitted-to-hospital-for-stroke",
          "Id": "IND20"
      },
      {
          "Url": "https://www.nice.org.uk/standards-and-indicators/ccgoisindicators/proportion-of-patients-with-atrial-fibrillation-on-anticoagulation-admitted-to-hospital-for-stroke",
          "Id": "IND101"
      }
  ]
}
