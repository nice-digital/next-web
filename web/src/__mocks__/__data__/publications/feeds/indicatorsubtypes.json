{"_links":{"self":[{"href":"/feeds/indicatorsubtypes"}]},"_embedded":{"nice.publications:indicator-sub-type-list":{"_links":{"self":[{}]},"_embedded":{"nice.publications:indicator-sub-type":[{"_links":{"self":[{}]},"ETag":null,"LastModified":"2022-07-12T00:00:00","Enabled":true,"Name":"Clinical commissioning group indicator","PluralName":"Clinical commissioning group indicators","IdentifierPrefix":"CCG"},{"_links":{"self":[{}]},"ETag":null,"LastModified":"2022-07-12T00:00:00","Enabled":true,"Name":"General practice indicator suitable for use in QOF","PluralName":"General practice indicators suitable for use in QOF","IdentifierPrefix":"GPIQ"},{"_links":{"self":[{}]},"ETag":null,"LastModified":"2022-07-12T00:00:00","Enabled":true,"Name":"General practice indicator suitable for use outside of QOF","PluralName":"General practice indicators suitable for use outside of QOF","IdentifierPrefix":"GPINQ"},{"_links":{"self":[{}]},"ETag":null,"LastModified":"2022-07-12T00:00:00","Enabled":true,"Name":"National library of quality indicators","PluralName":"National library of quality indicators","IdentifierPrefix":"NLQ"}]},"ETag":null}},"ETag":null,"LastModified":"2022-07-12T00:00:00"}