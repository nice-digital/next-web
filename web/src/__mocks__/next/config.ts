import nextConfig from "./../../../next.config";

const getConfig = (): unknown => ({
	publicRuntimeConfig: nextConfig.publicRuntimeConfig,
	serverRuntimeConfig: nextConfig.serverRuntimeConfig,
});

export default getConfig;
