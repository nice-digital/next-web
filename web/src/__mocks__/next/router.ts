export const useRouter = jest.fn().mockImplementation(() => {
	throw new Error("Provide a mock of useRouter for testing purposes");
});
