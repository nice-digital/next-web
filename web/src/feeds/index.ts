import { type Readable } from "stream";

import axios, { type AxiosInstance } from "axios";
import applyCaseMiddleware from "axios-case-converter";
import { camelCase } from "camel-case";

import { cache, getCacheKey } from "@/cache";

export const client: AxiosInstance = applyCaseMiddleware(
	axios.create({
		headers: { accept: "application/json" },
	}),
	{
		caseFunctions: {
			camel: (input, _options) => {
				// Strip out the prefix from all the embedded keys as it creates so much noise
				return camelCase(input.replace("nice.publications:", ""));
			},
		},
	}
);

/**
 * Gets the body of a feed directly from the back end system
 *
 * @param path The path of the feed to request
 * @returns The body of the feed
 */
export const getFeedBodyUnCached = async <TResponse>(
	origin: string,
	path: string,
	apiKey: string,
	acceptHeader = "application/json"
): Promise<TResponse> => {
	const { data } = await client.get<TResponse>(origin + path, {
		headers: {
			"Api-Key": apiKey,
			Accept: acceptHeader,
		},
		validateStatus: (status: number) => {
			// We don't want feed 404 responses to throw an error, so that we can show users a not found page rather than a server error.
			return (status >= 200 && status < 300) || status == 404;
		},
	});

	return data;
};

/**
 * Gets the body of a feed from the cache, if it exists.
 * If the item isn't in cache then it will call the `getUncachedAction` function to get fresh data.
 *
 * @param path The path of the feed endpoint to request
 * @param ttl The TTL (time to live) of the entry
 * @param getUncachedAction Function that gets the uncached feed directly from the back end.
 * @returns The body of the feed
 */
export const getFeedBodyCached = async <T>(
	groupCacheKey: string,
	path: string,
	ttl: number,
	getUncachedAction: () => Promise<T>
): Promise<T> =>
	cache.wrap<T>(getCacheKey(groupCacheKey.toLowerCase(), path.toLowerCase()), getUncachedAction, { ttl });

/**
 * Gets a response stream from a remote API endpoint, usually used for binary files e.g. PDFs (or mobi/epub etc).
 *
 * @param origin The base URL of the endpoint
 * @param path The path to the endpoint e.g. `/feeds/downloads/737585a0-dad7-4a37-875b-b30b09c3fdc3`
 * @param apiKey The API key for authentication against the API
 * @returns A readable response stream
 */
export const getResponseStream = async (
	origin: string,
	path: string,
	apiKey: string
): Promise<Readable> => {
	const { data } = await axios.get<Readable>(origin + path, {
		headers: {
			"Api-Key": apiKey,
		},
		responseType: "stream",
		maxBodyLength: Infinity,
	});
	return data;
};
