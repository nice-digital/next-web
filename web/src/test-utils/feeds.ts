import fs from "fs/promises";
import path from "path";

import MockAdapter from "axios-mock-adapter";

import { client } from "../feeds";

const mockDataBaseDirectory = path.resolve(
		__dirname,
		"../",
		"__mocks__",
		"__data__"
	),
	// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
	publicationsBaseUrl = new URL(process.env.SERVER_FEEDS_PUBLICATIONS_ORIGIN!),
	// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
	inDevBaseUrl = new URL(process.env.SERVER_FEEDS_INDEV_ORIGIN!);

export const axiosJSONMock = new MockAdapter(client, {
	onNoMatch: "throwException",
});

/**
 * Adds default JSON endpoint mocks for publications and indev feeds.
 *
 * Maps the incoming request URL to static JSON files on the file system in the mock data folder
 */
export const addDefaultJSONFeedMocks = (): MockAdapter =>
	axiosJSONMock.onGet().reply(async (config) => {
		if (!config.url) throw Error(`URL was empty in Axios mock`);

		const { host, pathname } = new URL(config.url),
			isHTML = config.headers?.Accept === "text/html";

		let filePath = mockDataBaseDirectory;

		if (host === publicationsBaseUrl.host) {
			filePath = path.join(filePath, "publications");
		} else if (host === inDevBaseUrl.host) {
			filePath = path.join(filePath, "inDev");
		} else
			throw Error(
				`Unknown URL host for mocking. Expected either ${publicationsBaseUrl.host} or ${inDevBaseUrl.host} but got ${host}`
			);

		filePath = path.join(
			filePath,
			pathname.toLowerCase() + (isHTML ? ".html" : ".json")
		);

		try {
			await fs.access(filePath);
		} catch {
			throw Error(`Could not find mock JSON file with path ${filePath}`);
		}

		const fileContents = await fs.readFile(filePath, "utf8");

		return [
			200,
			isHTML ? fileContents : !fileContents ? "" : JSON.parse(fileContents),
		];
	});
