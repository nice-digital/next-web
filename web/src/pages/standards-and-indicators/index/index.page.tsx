import { GetServerSideProps } from "next/types";

export const getServerSideProps: GetServerSideProps = async () => {
	return {
		redirect: {
			destination: "/indicators/published",
			permanent: true,
		},
	};
};

export default function OldIndicatorListPage(): void {
	// Default export to prevent next.js errors: we don't need an actual component
	// because the work is done in getServerSideProps to redirect
}
