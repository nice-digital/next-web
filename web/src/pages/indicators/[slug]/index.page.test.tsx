import {
	render,
	screen,
	waitFor,
	getDefaultNormalizer,
	within,
} from "@testing-library/react";
import MockAdapter from "axios-mock-adapter";
import { useRouter } from "next/router";

import { client } from "@/feeds/index";
import mockIndicatorSubTypes from "@/mockData/publications/feeds/products/indicator-sub-types.json";
import mockProduct from "@/mockData/publications/feeds/products/indicator.json";

import IndicatorsDetailsPage, {
	getServerSideProps,
	IndicatorsDetailsPageProps,
} from "./index.page";

import type { GetServerSidePropsContext } from "next";

const axiosMock = new MockAdapter(client, {
	onNoMatch: "throwException",
});

describe("/indicators/[slug].page", () => {
	const slug =
		"ind1001-test-indicator-ind-1001-the-percentage-of-patients-with-one-or-more-of-the-following-conditions-chd-atrial-fibrillation-chronic-heart-failure-stroke-or-tia-diabetes-or-dementia-with-a-fast-score-of-3-or-more-or-audit-c-score-of-5-or-more-in-the-preceding-2-years-who-have-received-brief-intervention-to-help-them-reduce-their-alcohol-related-risk-within-3-months-of-the-score-being-recorded";

	beforeEach(() => {
		(useRouter as jest.Mock).mockImplementation(() => ({}));
		axiosMock.reset();

		axiosMock.onGet(/\/feeds\/product\//).reply(200, mockProduct);

		axiosMock
			.onGet(/\/feeds\/indicatorsubtypes/)
			.reply(200, mockIndicatorSubTypes);

		jest.resetModules();
	});

	describe("IndicatorDetailPage", () => {
		let props: IndicatorsDetailsPageProps;
		beforeEach(async () => {
			const context = {
				params: { slug: slug },
			} as unknown as GetServerSidePropsContext;

			props = (
				(await getServerSideProps(context)) as {
					props: IndicatorsDetailsPageProps;
				}
			).props;
		});

		it("should match snapshot for main content", () => {
			render(<IndicatorsDetailsPage {...props} />);
			expect(document.body).toMatchSnapshot();
		});

		it("should render the page title with reversed breadcrumbs for SEO", async () => {
			render(<IndicatorsDetailsPage {...props} />);
			await waitFor(() => {
				expect(document.title).toEqual(
					`${mockProduct.Title} | Indicators | Standards and Indicators`
				);
			});
		});

		describe("Meta tags", () => {
			it("should have a link to the schema", async () => {
				render(<IndicatorsDetailsPage {...props} />);
				// eslint-disable-next-line testing-library/no-node-access
				const schemaLink = document.querySelector(`link[rel="schema.DCTERMS"]`);
				await waitFor(() => {
					expect(schemaLink).toBeInTheDocument();
				});

				expect(schemaLink).toHaveAttribute("href", "http://purl.org/dc/terms/");
			});

			it("should render the correct page meta tags for robots", async () => {
				render(<IndicatorsDetailsPage {...props} />);

				await waitFor(() => {
					expect(
						// eslint-disable-next-line testing-library/no-node-access
						document.querySelector(`meta[name="robots"]`)
					).toHaveAttribute("content", "index,follow");
				});
			});

			it("should render the correct page meta tags for description", async () => {
				render(<IndicatorsDetailsPage {...props} />);

				await waitFor(() => {
					expect(
						// eslint-disable-next-line testing-library/no-node-access
						document.querySelector(`meta[name="description"]`)
					).toHaveAttribute("content", mockProduct.MetaDescription);
				});
			});

			it("should render the correct page meta tags for DCTERMS.issued", async () => {
				render(<IndicatorsDetailsPage {...props} />);

				await waitFor(() => {
					expect(
						// eslint-disable-next-line testing-library/no-node-access
						document.querySelector(`meta[name="DCTERMS.issued"]`)
					).toHaveAttribute("content", mockProduct.PublishedDate);
				});
			});

			it("should render the correct page meta tags for DCTERMS.modified", async () => {
				render(<IndicatorsDetailsPage {...props} />);

				await waitFor(() => {
					expect(
						// eslint-disable-next-line testing-library/no-node-access
						document.querySelector(`meta[name="DCTERMS.modified"]`)
					).toHaveAttribute("content", mockProduct.LastMajorModificationDate);
				});
			});

			it("should render multiple meta tags for DCTERMS.type", async () => {
				render(<IndicatorsDetailsPage {...props} />);
				// eslint-disable-next-line testing-library/no-node-access
				const typeMetaTags = document.querySelectorAll(
					`meta[name="DCTERMS.type"]`
				);
				await waitFor(() => {
					expect(typeMetaTags).toHaveLength(2);
				});
				expect(typeMetaTags[0]).toHaveAttribute(
					"content",
					"Clinical commissioning group indicator"
				);
				expect(typeMetaTags[1]).toHaveAttribute(
					"content",
					"General practice indicator suitable for use in QOF"
				);
			});

			it("should render the correct page meta tags for DCTERMS.identifier", async () => {
				render(<IndicatorsDetailsPage {...props} />);

				await waitFor(() => {
					expect(
						// eslint-disable-next-line testing-library/no-node-access
						document.querySelector(`meta[name="DCTERMS.identifier"]`)
					).toHaveAttribute("content", mockProduct.Id);
				});
			});
		});

		describe("PageHeader", () => {
			it("should render meta data id in uppercase", () => {
				render(<IndicatorsDetailsPage {...props} />);
				expect(
					screen.getByText(mockProduct.Id, {
						selector: ".page-header__metadata li",
					})
				).toBeInTheDocument();
			});

			it.each([
				["NICE indicator"],
				["IND1001"],
				["Published: 8 September 2022"],
				["Last updated: 12 October 2022"],
			])("should render a %s page header meta element", (metaContent) => {
				render(
					<IndicatorsDetailsPage
						{...props}
						product={{
							...props.product,
							lastMajorModificationDate: "2022-10-12",
						}}
					/>
				);

				expect(
					screen.getByText(
						(content, element) => {
							return (
								getDefaultNormalizer()(element?.textContent as string) ==
								metaContent
							);
						},
						{
							selector: ".page-header__metadata li",
						}
					)
				);
			});

			it("should not render last updated date if published date == lastModified date", () => {
				render(
					<IndicatorsDetailsPage
						{...props}
						product={{
							...props.product,
							lastMajorModificationDate: props.product.publishedDate,
						}}
					/>
				);

				expect(screen.queryByText("Last updated:")).not.toBeInTheDocument();
			});

			it("should render last updated date if published date !== lastModified date", () => {
				render(
					<IndicatorsDetailsPage
						{...props}
						product={{
							...props.product,
							lastMajorModificationDate: "2022-10-12",
						}}
					/>
				);

				expect(screen.getByText("Last updated:")).toBeInTheDocument();
			});

			it("should render 'Published' date page header lead meta element in the correct format", () => {
				render(<IndicatorsDetailsPage {...props} />);
				const publishedDateEl = screen.getByText("8 September 2022", {
					selector: "time",
				});
				expect(publishedDateEl).toBeInTheDocument();
			});

			it("should render 'Published' date page header lead meta element with correctly formatted datetime attribute", () => {
				render(<IndicatorsDetailsPage {...props} />);
				const publishedDateEl = screen.getByText("8 September 2022", {
					selector: "time",
				});
				expect(publishedDateEl).toHaveAttribute("datetime", "2022-09-08");
			});

			it("should render 'Last updated' date page header lead meta element in the correct format", () => {
				render(
					<IndicatorsDetailsPage
						{...props}
						product={{
							...props.product,
							lastMajorModificationDate: "2022-10-12",
						}}
					/>
				);
				const publishedDateEl = screen.getByText("12 October 2022", {
					selector: "time",
				});
				expect(publishedDateEl).toBeInTheDocument();
			});

			it("should render 'Last updated' date page header lead meta element with correctly formatted datetime attribute", () => {
				render(
					<IndicatorsDetailsPage
						{...props}
						product={{
							...props.product,
							lastMajorModificationDate: "2022-10-12",
						}}
					/>
				);
				const publishedDateEl = screen.getByText("12 October 2022", {
					selector: "time",
				});
				expect(publishedDateEl).toHaveAttribute("datetime", "2022-10-12");
			});
		});

		describe("Chapter menu", () => {
			it("should render overview chapter link when summary provided", () => {
				render(
					<IndicatorsDetailsPage
						{...props}
						product={{
							...props.product,
							summary: mockProduct.Summary,
						}}
					/>
				);

				const publicationsChapterMenu = screen.getByRole("region", {
					name: "Chapters",
				});

				expect(
					within(publicationsChapterMenu).getByText("Overview")
				).toBeInTheDocument();
			});

			it("should not render overview chapter link when no summary provided", () => {
				render(
					<IndicatorsDetailsPage
						{...props}
						product={{
							...props.product,
							summary: null,
						}}
					/>
				);

				const publicationsChapterMenu = screen.getByRole("region", {
					name: "Chapters",
				});

				expect(
					within(publicationsChapterMenu).queryByText("Overview")
				).not.toBeInTheDocument();
			});
		});
	});

	describe("getServerSideProps", () => {
		it("should return a correct props when supplied with an id", async () => {
			const result = await getServerSideProps({
				params: { slug },
			} as unknown as GetServerSidePropsContext);

			expect(result).toStrictEqual({
				props: {
					slug: slug,
					product: expect.objectContaining({ id: "IND1001" }),
					indicatorSubTypes: [
						expect.objectContaining({
							name: "Clinical commissioning group indicator",
						}),
						expect.objectContaining({
							name: "General practice indicator suitable for use in QOF",
						}),
						expect.objectContaining({
							name: "General practice indicator suitable for use outside of QOF",
						}),
						expect.objectContaining({
							name: "National library of quality indicators",
						}),
					],
					pdfDownloadPath: `/indicators/${slug}/IND1001.pdf`,
				},
			});
		});

		describe("Redirects", () => {
			it("should return permanent redirect object URL with incorrect title", async () => {
				const redirectResult = await getServerSideProps({
					params: { slug: "ind1001-incorrect-slug-title" },
				} as unknown as GetServerSidePropsContext);

				expect(redirectResult).toStrictEqual({
					redirect: {
						destination: `/indicators/${slug}`,
						permanent: true,
					},
				});
			});
			it("should return notFound if Id doesn't exist", async () => {
				const redirectResult = await getServerSideProps({
					params: { slug: "nonExistingId99-test-title-1" },
				} as unknown as GetServerSidePropsContext);

				expect(redirectResult).toStrictEqual({ notFound: true });
			});

			it("should return notFound if an unhyphenated slug is used", async () => {
				const redirectResult = await getServerSideProps({
					params: { slug: "slugwithouthyphenation" },
				} as unknown as GetServerSidePropsContext);

				expect(redirectResult).toStrictEqual({ notFound: true });
			});
		});
	});
});
