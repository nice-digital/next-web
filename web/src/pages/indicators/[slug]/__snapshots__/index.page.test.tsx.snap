// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/indicators/[slug].page IndicatorDetailPage should match snapshot for main content 1`] = `
<body>
  <div>
    <nav
      aria-label="Breadcrumbs"
      data-component="breadcrumbs"
      role="navigation"
    >
      <ol
        class="breadcrumbs"
      >
        <li
          class="breadcrumbs__crumb"
        >
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="breadcrumbs__crumb"
        >
          <a
            href="/standards-and-indicators/indicators"
          >
            Indicators
          </a>
        </li>
        <li
          class="breadcrumbs__crumb"
        >
          <span>
            NG100
          </span>
        </li>
      </ol>
    </nav>
    <script
      type="application/ld+json"
    >
      {
  "@context": "http://schema.org",
  "@type": "BreadcrumbList",
  "itemListElement": [
    {
      "@type": "ListItem",
      "position": 1,
      "item": {
        "@id": "/",
        "name": "Home"
      }
    },
    {
      "@type": "ListItem",
      "position": 2,
      "item": {
        "@id": "/standards-and-indicators/indicators",
        "name": "Indicators"
      }
    },
    {
      "@type": "ListItem",
      "position": 3,
      "item": {
        "name": "NG100"
      }
    }
  ]
}
    </script>
    <div
      class="page-header"
      data-component="page-header"
      id="content-start"
    >
      <h1
        class="page-header__heading page-header__heading--alt"
      >
        Rheumatoid arthritis in adults: management
      </h1>
      <ul
        class="page-header__metadata"
      >
        <li>
          NICE guideline
        </li>
        <li>
          NG100
        </li>
        <li>
          Published:
          <time
            datetime="2018-07-11"
          >
             
            11 July 2018
          </time>
        </li>
        <li>
          Last updated:
          <time
            datetime="2020-10-12"
          >
             
             
            12 October 2020
          </time>
        </li>
      </ul>
    </div>
    <nav
      class="horizontal-nav"
    >
      <ul
        class="horizontal-nav__list"
      >
        <li
          class="horizontal-nav__item"
          data-component="horizontal-nav"
        >
          <a
            aria-current="true"
            class="horizontal-nav__link"
            href="/guidance/ng100"
          >
            Indicator
          </a>
        </li>
        <li
          class="horizontal-nav__item"
          data-component="horizontal-nav"
        >
          <a
            aria-current="false"
            class="horizontal-nav__link"
            href="/guidance/ng100/resources"
          >
            Tools and resources
          </a>
        </li>
        <li
          class="horizontal-nav__item"
          data-component="horizontal-nav"
        >
          <a
            aria-current="false"
            class="horizontal-nav__link"
            href="/guidance/ng100/evidence"
          >
            Evidence
          </a>
        </li>
        <li
          class="horizontal-nav__item"
          data-component="horizontal-nav"
        >
          <a
            aria-current="false"
            class="horizontal-nav__link"
            href="/guidance/ng100/history"
          >
            History
          </a>
        </li>
      </ul>
    </nav>
    <div
      class="grid grid--loose"
    >
      <section
        aria-label="Chapters"
        data-component="grid-item"
        data-g="12 md:4 lg:3"
      >
        <a
          aria-label="Download indicator PDF file"
          class="btn btn--cta download"
          data-component="button--cta"
          href="/guidance/ng100/NG100-20220530.pdf"
          target="_blank"
        >
          Download indicator
           (PDF)
        </a>
        <nav
          aria-label="Chapter pages"
          class="stacked-nav"
          data-component="stacked-nav"
        >
          <ul
            class="stacked-nav__list"
          >
            <li
              class="stacked-nav__list-item"
            >
              <a
                aria-current="true"
                href="/guidance/ng100"
              >
                <span
                  class="stacked-nav__content-wrapper"
                >
                  <span>
                    Overview
                  </span>
                </span>
              </a>
            </li>
            <li
              class="stacked-nav__list-item"
            >
              <a
                aria-current="false"
                href="/guidance/ng100/chapter/recommendations"
              >
                <span
                  class="stacked-nav__content-wrapper"
                >
                  <span>
                    Recommendations
                  </span>
                </span>
              </a>
            </li>
            <li
              class="stacked-nav__list-item"
            >
              <a
                aria-current="false"
                href="/guidance/ng100/chapter/recommendations-for-research"
              >
                <span
                  class="stacked-nav__content-wrapper"
                >
                  <span>
                    Recommendations for research
                  </span>
                </span>
              </a>
            </li>
            <li
              class="stacked-nav__list-item"
            >
              <a
                aria-current="false"
                href="/guidance/ng100/chapter/rationale-and-impact"
              >
                <span
                  class="stacked-nav__content-wrapper"
                >
                  <span>
                    Rationale and impact
                  </span>
                </span>
              </a>
            </li>
            <li
              class="stacked-nav__list-item"
            >
              <a
                aria-current="false"
                href="/guidance/ng100/chapter/context"
              >
                <span
                  class="stacked-nav__content-wrapper"
                >
                  <span>
                    Context
                  </span>
                </span>
              </a>
            </li>
            <li
              class="stacked-nav__list-item"
            >
              <a
                aria-current="false"
                href="/guidance/ng100/chapter/finding-more-information-and-committee-details"
              >
                <span
                  class="stacked-nav__content-wrapper"
                >
                  <span>
                    Finding more information and committee details
                  </span>
                </span>
              </a>
            </li>
            <li
              class="stacked-nav__list-item"
            >
              <a
                aria-current="false"
                href="/guidance/ng100/chapter/update-information"
              >
                <span
                  class="stacked-nav__content-wrapper"
                >
                  <span>
                    Update information
                  </span>
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </section>
      <section
        data-component="grid-item"
        data-g="12 md:8 lg:9"
      >
        <div
          class="summary"
        >
          <p
            class="lead"
          >
            This guideline covers diagnosing and managing rheumatoid arthritis. It aims to improve quality of life by ensuring that people with rheumatoid arthritis have the right treatment to slow the progression of their condition and control their symptoms. People should also have rapid access to specialist care if their condition suddenly worsens.
          </p>
          

          <p>
            NICE has also produced technology appraisal guidance on drug treatment for rheumatoid arthritis.
          </p>
          

          <p>
            In 
            <strong>
              October 2020
            </strong>
            , we amended the ‘treat to target’ recommendations to clarify that multiple disease-modifying anti-rheumatic drugs can be offered one after the other to achieve treatment targets.
          </p>
          

          <h2>
            Recommendations
          </h2>
          

          <p>
            This guideline includes new and updated recommendations on:
          </p>
          

          <ul>
            

            <li>
              <a
                href="https://www.nice.org.uk/guidance/ng100/chapter/Recommendations#investigations-following-diagnosis"
              >
                investigations following diagnosis
              </a>
            </li>
            

            <li>
              <a
                href="https://www.nice.org.uk/guidance/ng100/chapter/Recommendations#treat-to-target-strategy"
              >
                treat-to-target strategy
              </a>
               and 
              <a
                href="http://www.nice.org.uk/guidance/ng100/chapter/Recommendations#initial-pharmacological-management"
              >
                initial pharmacological management
              </a>
            </li>
            

            <li>
              <a
                href="https://www.nice.org.uk/guidance/ng100/chapter/Recommendations#symptom-control"
              >
                symptom control
              </a>
               and 
              <a
                href="http://www.nice.org.uk/guidance/ng100/chapter/Recommendations#monitoring"
              >
                monitoring
              </a>
            </li>
            

          </ul>
          

          <p>
            It also includes recommendations on:
          </p>
          

          <ul>
            

            <li>
              <a
                href="http://www.nice.org.uk/guidance/ng100/chapter/Recommendations#investigations"
              >
                investigations for diagnosis
              </a>
               and 
              <a
                href="http://www.nice.org.uk/guidance/ng100/chapter/Recommendations#referral-from-primary-care"
              >
                referral from primary care
              </a>
            </li>
            

            <li>
              <a
                href="http://www.nice.org.uk/guidance/ng100/chapter/Recommendations#non-pharmacological-management"
              >
                non-pharmacological management
              </a>
               and 
              <a
                href="http://www.nice.org.uk/guidance/ng100/chapter/Recommendations#the-multidisciplinary-team"
              >
                the multidisciplinary team
              </a>
            </li>
            

            <li>
              <a
                href="http://www.nice.org.uk/guidance/ng100/chapter/Recommendations#communication-and-education"
              >
                communication and education
              </a>
            </li>
            

            <li>
              <a
                href="http://www.nice.org.uk/guidance/ng100/chapter/Recommendations#timing-and-referral-for-surgery"
              >
                timing and referral for surgery
              </a>
            </li>
            

          </ul>
          

          <h2>
            Who is it for?
          </h2>
          

          <ul>
            

            <li>
              Healthcare professionals
            </li>
            

            <li>
              Commissioners and providers
            </li>
            

            <li>
              People with rheumatoid arthritis and their families and carers
            </li>
            

          </ul>
          

          <h2>
            Guideline development process
          </h2>
          

          <p>
            <a
              href="https://www.nice.org.uk/about/what-we-do/our-programmes/nice-guidance/nice-guidelines/how-we-develop-nice-guidelines"
            >
              How we develop NICE guidelines
            </a>
          </p>
          

          <p>
            This guideline updates and replaces NICE guideline CG79 (February 2009).
          </p>
        </div>
        <div
          class="prev-next"
        >
          <nav
            aria-label="Previous and next pages"
          >
            <a
              class="prev-next__link"
              href="/guidance/ng100/chapter/recommendations"
            >
              <span
                class="prev-next__link-intro"
              >
                Next page
              </span>
              <span
                class="prev-next__link-text"
              >
                Recommendations
              </span>
            </a>
          </nav>
        </div>
      </section>
    </div>
  </div>
</body>
`;

exports[`/indicators/[slug].page getServerSideProps should return a correct props when supplied with a valid slug 1`] = `
Object {
  "redirect": Object {
    "destination": "/guidance/ng100",
    "permanent": true,
  },
}
`;
